ocean_lib/assets/v4/asset.py

```diff
@@ -0,0 +1,244 @@
+
+    @property
+    def allowed_addresses(self) -> list:
+        """Lists addresses that are explicitly allowed in credentials."""
+        manager = AddressCredential(self)
+        return manager.get_addresses_of_class("allow")
+
+    @property
+    def denied_addresses(self) -> list:
+        """Lists addresses that are explicitly denied in credentials."""
+        manager = AddressCredential(self)
+        return manager.get_addresses_of_class("deny")
+
+    def add_address_to_allow_list(self, address: str) -> None:
+        """Adds an address to allowed addresses list."""
+        manager = AddressCredential(self)
+        manager.add_address_to_access_class(address, "allow")
+
+    def add_address_to_deny_list(self, address: str) -> None:
+        """Adds an address to the denied addresses list."""
+        manager = AddressCredential(self)
+        manager.add_address_to_access_class(address, "deny")
+
+    def remove_address_from_allow_list(self, address: str) -> None:
+        """Removes address from allow list (if it exists)."""
+        manager = AddressCredential(self)
+        manager.remove_address_from_access_class(address, "allow")
+
+    def remove_address_from_deny_list(self, address: str) -> None:
+        """Removes address from deny list (if it exists)."""
+        manager = AddressCredential(self)
+        manager.remove_address_from_access_class(address, "deny")
+
+    @classmethod
+    def from_dict(cls, dictionary: dict) -> "V4Asset":
+        """Import a JSON dict into this Asset."""
+        values = copy.deepcopy(dictionary)
+        id_key = "id" if "id" in values else "_id"
+        did = values.pop(id_key)
+        chain_id = values.pop("chainId")
+        context = values.pop("@context")
+
+        if "metadata" in values:
+            metadata = values.pop("metadata")
+        if "services" in values:
+            services = []
+            for value in values.pop("services"):
+                if isinstance(value, str):
+                    value = json.loads(value)
+
+                service = V4Service.from_json(value)
+                services.append(service)
+        if "credentials" in values:
+            credentials = values.pop("credentials")
+        if "nft" in values:
+            nft = values.pop("nft")
+        if "datatokens" in values:
+            data_tokens = values.pop("datatokens")
+        if "event" in values:
+            event = values.pop("event")
+        if "stats" in values:
+            stats = values.pop("stats")
+        return cls(
+            did,
+            context,
+            chain_id,
+            metadata,
+            services,
+            credentials,
+            nft,
+            data_tokens,
+            event,
+            stats,
+        )
+
+    def as_dictionary(self) -> dict:
+        """
+        Return the DDO as a JSON dict.
+
+        :return: dict
+        """
+
+        data = {
+            "@context": ["https://w3id.org/did/v1"],
```

Comment:
only hardcode this if it is somehow missing. if there is a `@context` property on the object, please use that one. Otherwise fall back to this.